portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,67 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  ArrayList<Comment> comments = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String json = this.convertToJson(this.comments);
+    System.out.println(json);
+    
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = getParameter(request, "name", "Anonymous");
+    String body = getParameter(request, "body", "");
+
+    Comment newComment = new Comment(name, body);
+    this.comments.add(newComment);
+
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.sendRedirect("/#comments");
+
+  }
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  private String convertToJsonold(ArrayList<Comment> comments) {
+    String out = "{\"comments\":[";
+    boolean first = true;
+    for(Comment comment: comments){
```

Comment:
Consider using something like String.join("," commets), see https://stackoverflow.com/questions/599161/best-way-to-convert-an-arraylist-to-a-string/23183963 (the second answer, because we work on Java 8 or newer).

Reply:
Thanks for the info!!!